modules/core/core-content/src/main/java/com/enonic/xp/core/impl/project/ProjectPermissionsContextManagerImpl.java

```diff
@@ -124,27 +124,18 @@ public boolean hasAnyProjectPermission( final ProjectName projectName, final Aut
                                        ProjectRole.VIEWER ) );
     }
 
-    private boolean hasAdminProjectPermission( final ProjectName projectName, final AuthenticationInfo authenticationInfo )
-    {
-        return hasPermissions( projectName, authenticationInfo, Set.of( ProjectRole.OWNER ) );
-    }
-
-    private boolean hasEditorProjectPermission( final ProjectName projectName, final AuthenticationInfo authenticationInfo )
-    {
-        return hasPermissions( projectName, authenticationInfo, Set.of( ProjectRole.EDITOR ) );
-    }
-
-    private boolean hasAuthorProjectPermission( final ProjectName projectName, final AuthenticationInfo authenticationInfo )
+    @Override
+    public boolean hasAnyProjectPermission( final ProjectName projectName, final AuthenticationInfo authenticationInfo,
+                                            final Set<ProjectRole> projectRoles )
     {
-        return hasPermissions( projectName, authenticationInfo, Set.of( ProjectRole.AUTHOR ) );
+        return hasPermissions( projectName, authenticationInfo, projectRoles );
     }
 
-    private boolean hasContributorProjectPermission( final ProjectName projectName, final AuthenticationInfo authenticationInfo )
+    private boolean hasOwnerProjectPermission( final ProjectName projectName, final AuthenticationInfo authenticationInfo )
     {
-        return hasPermissions( projectName, authenticationInfo, Set.of( ProjectRole.CONTRIBUTOR ) );
+        return hasPermissions( projectName, authenticationInfo, Set.of( ProjectRole.OWNER ) );
```

Comment:
We decided to use hasRole method of AuthInfo but it means that internally service must know how to construct project role names correctly - which should be done in another class (to share logic)